### DISCUSSION TITLE
use external modules library in driver using CMake

### DISCUSSION THREAD

**Discussion by dornbirndevelops** [zephyrproject-rtos/zephyr#43871] [Q&A] [answered]
Hi, quick question regarding library integration.
I'm developing a sensor driver `foo_impl.c` that is supposed to use symbols declared in `foo.h` which in turn are implemented by `foo.c`.

```
zephyrproject/
├─ modules/lib/foo/
│  ├─ foo.h
│  ├─ foo.c
│  ├─ CMakeLists.txt
│  ├─ zephyr/
│  │  ├─ module.yml
├─ zephyr/drivers/sensor/foo/
│  ├─ foo_impl.c
```

How does the implementation of `CMakeLists.txt` need to look like, so I can use the module library in Zephyr?

**Comment by boaks** [comment, marked as answer]
Maybe related to [Best practice for "external libraries" and cmake](https://github.com/zephyrproject-rtos/zephyr/discussions/43650)?

**Reply by dornbirndevelops** [reply]
yes, I also stumbled across your discussion. We can continue there
